### PR TITLE
AP_Camera: increase camera triggers distance accuracy (2)

### PR DESCRIPTION
**Pull request by mirkix**
Rebase to master and cleanup of https://github.com/ArduPilot/ardupilot/pull/14658.

This PR increase the camera trigger accuracy when using automatic camera trigger with https://ardupilot.org/copter/docs/parameters.html#cam-trigg-dist-camera-trigger-distance. Until now a new trigger will be checked with each GPS update (~10Hz), after this PR the trigger will be checked with 50Hz.
With ~10Hz and flight speed of 15m/s we only check about each 1,5m if we have to trigger the camera again, this should be improved.

### CODE REVIEW

**Comment by mirkix** [comment]
@khancyr This PR will also increase the camera triggers distance accuracy for Plane.

**Comment by khancyr** [comment]
oh ... my bad, I miss it !

File: libraries/AP_Camera/AP_Camera.cpp [hunk 1]
```diff
@@ -476,6 +476,8 @@ void AP_Camera::take_picture()
  */
 void AP_Camera::update_trigger()
 {
+    update();
```
Comment (rmackay9) [inline libraries/AP_Camera/AP_Camera.cpp:479]: It seems odd to me that update() is called from within update_trigger().  I think instead we should make update_trigger private and call it from within the public "update".

**Comment by mirkix** [comment]
@rmackay9 Hello Randy, can you check again please.

**Comment by rmackay9** [comment]
Merged, thanks.  I've created an issue for the semaphore failure which seems unrelated. https://github.com/ArduPilot/ardupilot/issues/14816

**Pull request merged**